### PR TITLE
Better rdoc for public APIs

### PR DESCRIPTION
**Pull request by mperham**
Improve the RDoc for "sidekiq/api".

A good cheatsheet for YARD syntax:

https://gist.github.com/donnoman/562875

cc @SamArdrey

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
-RDoc::Task.new do |rdoc|
-  rdoc.main = "docs/rdoc.rdoc"
-  rdoc.rdoc_files.include("docs/rdoc.rdoc",
-    "lib/sidekiq/api.rb",
-    "lib/sidekiq/client.rb",
-    "lib/sidekiq/worker.rb",
-    "lib/sidekiq/job.rb")
+YARD::Rake::YardocTask.new do |yard|
+  # keeping this in here for now, but code can be deleted once we are ready
+  # to push. .yardopts takes care of this.
```
Comment (SamArdrey) [inline Rakefile:9]: @mperham just leaving a comment here for you so it's not overlooked when the come comes to mergs :)

File: lib/sidekiq/api.rb [hunk 2]
```diff
@@ -5,7 +5,9 @@
 require "zlib"
 require "base64"
 
+# @api public
 module Sidekiq
+  # @api private
```
Comment (SamArdrey) [inline lib/sidekiq/api.rb:10]: This is private till we have more documentation on stats. Right now, we are a bit slim

**Comment by SamArdrey** [comment]
@mperham this is ready for review. I still have a lot of work to do to get it perfect, but it's good enough for what we need. The classes show up on the left side as well. I'd recommend testing it to make sure it's up to your standards. I have time in the morning to do any changes you need before we meet.

**Comment by mperham** [comment]
I still see nothing in the left sidebar for Classes.
Screen Shot 2022-06-15 at 8 09 35 AM

**Comment by SamArdrey** [comment]
Oh damn, I had it working there for a while. Lemme look into it

**Comment by SamArdrey** [comment]
@mperham Got the classes working for only the files we want. Will keep perfecting this.

**Comment by mperham** [comment]
All done, you may merge at your convenience @SamArdrey.

**Pull request merged**